### PR TITLE
Remove test .pfx file from repo. Generate temp cert as part of the build.

### PR DESCRIPTION
**Pull request by kmahone**
It is not best practice to include a checked in .pfx in the git repo.

We need a .pfx file to test sign the test app appx files that we create. An appx must be signed with something for it to be able to be installed. We want to maintain the ability for someone to just clone the repo and build the .sln without having to do any special steps to configure their environment. So, we now generate a new self-signed .pfx file as part of the build to use for signing with GenerateTestPfx.ps1. 

As part of this change, I have also updated the published identity for the test apps from 'Microsoft' to 'WinUITest' which is more appropriate.

For the Microsoft.UI.Xaml Framework package, we need to keep the publisher identity as 'Microsoft', so we generate a separate temp self-signed pfx for that. 

None of this has any bearing on what we ship in official release builds. For these we sign everything using an internal signing system and not the temp self-signed pfx files that we use for testing.

### CODE REVIEW

File: GenerateTestPfx.ps1 [hunk 1]
```diff
@@ -0,0 +1,12 @@
+$CertificateFriendlyName = "WinUITest"
+$Publisher = "CN=WinUITest"
+
+$cert = New-SelfSignedCertificate -Type Custom `
+    -Subject $Publisher `
+    -KeyUsage DigitalSignature `
+    -FriendlyName $CertificateFriendlyName `
+    -CertStoreLocation "Cert:\CurrentUser\My" `
+    -TextExtension @("2.5.29.37={text}1.3.6.1.5.5.7.3.3", "2.5.29.19={text}")
```
Comment (llongley) [inline GenerateTestPfx.ps1:9]: Does this mean anything, or is this just a magic incantation that works?

File: build/Helix/scripts/TestPass-PreRun.ps1 [hunk 2]
```diff
@@ -21,7 +21,7 @@ function UninstallTestApps {
         # which prevents other versions of the app from being installed.
         # To handle this, we can directly call Remove-AppxPackage against the full name of the package. However, without
         # Get-AppxPackage to find the PackageFullName, we just have to manually construct the name.
-        $packageFullName = "$($pkgName)_1.0.0.0_$($Platform)__8wekyb3d8bbwe" 
+        $packageFullName = "$($pkgName)_1.0.0.0_$($Platform)__6f07fta6qpts2"
```
Comment (llongley) [inline build/Helix/scripts/TestPass-PreRun.ps1:24]: Is this hash (I'm assuming this is a hash) still going to be consistent now that we're generating a new PFX on the fly?

**Review by llongley — APPROVED** [review]

File: build/Helix/scripts/TestPass-PreRun.ps1 (hunk 2 above)
Comment (kmahone) [reply build/Helix/scripts/TestPass-PreRun.ps1:24]: Yes, it is consistent. It is based on the publisher name ("WinUITest").

File: GenerateTestPfx.ps1 (hunk 1 above)
Comment (kmahone) [reply GenerateTestPfx.ps1:9]: It's pretty opaque alright. Here is where I got it from:
https://docs.microsoft.com/en-us/windows/msix/package/create-certificate-package-signing#create-a-self-signed-certificate

**Pull request merged**